commit 36829f2454ec7da42c400dd260a43f4a221f9124 (refs/changes/91/3291/3)
Author: Rob Johnston <rob.johnston@joyent.com>
Date:   2018-02-02T20:39:21+00:00 (1 year, 8 months ago)
    
    OS-6129 fmadm reports faults without message summary
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/cmd/fm/modules/common/eversholt/fme.c b/usr/src/cmd/fm/modules/common/eversholt/fme.c
index 096caa73db..553db71b6c 100644
--- a/usr/src/cmd/fm/modules/common/eversholt/fme.c
+++ b/usr/src/cmd/fm/modules/common/eversholt/fme.c
@@ -22,6 +22,7 @@
 /*
  * Copyright (c) 2003, 2010, Oracle and/or its affiliates. All rights reserved.
  * Copyright 2012 Milan Jurik. All rights reserved.
+ * Copyright (c) 2018, Joyent, Inc.
  *
  * fme.c -- fault management exercise module
  *
@@ -350,6 +351,7 @@ newfme(const char *e0class, const struct ipath *e0ipp, fmd_hdl_t *hdl,
 			out(O_ALTFP|O_VERB2, "Unable to map \"%s\" ereport "
 			    "to component path, but silent discard allowed.",
 			    e0class);
+			fmd_case_close(hdl, fmcase);
 		} else {
 			Undiag_reason = UD_VAL_BADEVENTPATH;
 			(void) nvlist_lookup_nvlist(nvl, FM_EREPORT_DETECTOR,
